modules/ROOT/pages/cross_cutting/test_automation/sub_system_tests.adoc

```diff
@@ -0,0 +1,50 @@
+== Functional Subsystem- and System-Tests (API + UI)
+API test in this section means a functional test (step) that uses the API calls to trigger or check functionality of the fully integrated system under test.
+This is a difference to unit testing of a method that exposes an endpoint via REST.
+
+Typical Tool: RestAssured
+TODO: example
+
+=== Use BDD to develop test scenarios
+Typical Tool: Cucumber
+
+==== BDD and bad good practices
+https://automationpanda.com/2017/01/30/bdd-101-writing-good-gherkin/
+
+Most important:
+
+* Collaborate with business, dev and test
+* Short scenarios (leave out irrelevant data, combine UI steps)
+* Do NOT use Generic UI steps
+
+=== Mix API and UI steps
+Just because your expected result can only be checked in the UI does not mean your whole test must be executed via UI.
+You can use the API to create you preconditions and the perform some step in the UI.
+This makes the complete test faster and more reliable.
+
+=== Hide API authentication at a central point
+
+=== Encapsulate UI technology specific logic in page objects
+
+=== Reflect UI components in your test code
+
+=== Add screenshots to the result report
+
+=== Mock external system that are not in your scope
+Typical tool: Wiremock
+
+
+=== Keep references to data in the database at a central place
+
+=== Use test data builders with defaults for complex structures
+
+=== Prevent explicit waiting
+
+Selenium implicit wait
+
+Typical tool: Awaitility
+
+=== Apply a rerun for failed tests
+
+Cucumber, Maven surefire features
+
```

Comment:
Maybe additionally:
- Combination with Test Management e.g. in Jira / Xray for better collaboration with BU